### PR TITLE
Improve hucon image generation and some bug fixes

### PR DESCRIPTION
**Pull request by JonasTrampe**

### CODE REVIEW

File: python_lib/Eye.py [hunk 1]
```diff
@@ -12,12 +12,12 @@ class Eye(object):
     """ Eye driver class.
     """
 
-    RGB = [2, 1, 0]
-    RBG = [2, 0, 1]
-    GBR = [1, 0, 2]
-    GRB = [1, 2, 0]
-    BGR = [0, 1, 2]
-    BRG = [0, 2, 1]
+    RGB = [1, 2, 0]
```
Comment (SaschaMzH) [inline python_lib/Eye.py]: Have you tested these changes?

Comment (JonasTrampe) [reply python_lib/Eye.py]: No. Was split into another branch.

File: webserver/HuConJsonRpc.py [hunk 2]
```diff
@@ -63,11 +67,16 @@ def __init__(self):
...
+            os.makedirs(self._CODE_ROOT)
+
         print('%s v. %s' % (self._SERVER_NAME, self._version))
-        print('Code path: \'%s\'' % self._CODE_ROOT)
+        print('Custom code path: \'%s\'' % self._CODE_ROOT)
+        print('Example code path: \'%s\'' % self._EXAMPLE_ROOT)
 
     def handle_control(self, rpc_request):
-        """ Handle the JSON RPC request.
+        """
```
Comment (SaschaMzH) [inline webserver/HuConJsonRpc.py]: Please use the same comment mode over the whole file.

File: webserver/HuConJsonRpc.py [hunk 3]
```diff
@@ -199,13 +208,21 @@ def _get_file_list(self, rpc_request):
...
+            example_folder = os.path.join(self._EXAMPLE_ROOT, rpc_request['params'].strip('/\\'))
 
+            files_usercode = []
             rpc_response = self._get_rpc_response(rpc_request['id'])
-            rpc_response['result'] = os.listdir(code_folder)
+            if os.path.exists(code_folder):
+                files_usercode = os.listdir(code_folder)
+                files_usercode.sort()
+            files_examples = os.listdir(example_folder)
+            files_examples.sort()
```
Comment (SaschaMzH) [inline webserver/HuConJsonRpc.py]: Do you need to sort here?

File: webserver/HuConJsonRpc.py [hunk 4]
```diff
@@ -199,13 +208,21 @@ def _get_file_list(self, rpc_request):
...
         try:
             code_folder = os.path.join(self._CODE_ROOT, rpc_request['params'].strip('/\\'))
+            example_folder = os.path.join(self._EXAMPLE_ROOT, rpc_request['params'].strip('/\\'))
 
+            files_usercode = []
             rpc_response = self._get_rpc_response(rpc_request['id'])
-            rpc_response['result'] = os.listdir(code_folder)
+            if os.path.exists(code_folder):
+                files_usercode = os.listdir(code_folder)
+                files_usercode.sort()
```
Comment (SaschaMzH) [inline webserver/HuConJsonRpc.py]: Do you need to sort here?

File: webserver/HuConJsonRpc.py [hunk 5]
```diff
@@ -199,13 +208,21 @@ def _get_file_list(self, rpc_request):
         """
         try:
             code_folder = os.path.join(self._CODE_ROOT, rpc_request['params'].strip('/\\'))
+            example_folder = os.path.join(self._EXAMPLE_ROOT, rpc_request['params'].strip('/\\'))
```
Comment (SaschaMzH) [inline webserver/HuConJsonRpc.py]: What happen if you create an "example" folder inside the code folder?

**Review by SaschaMzH — CHANGES_REQUESTED** [review]
I have still questions about the implementation:

- What happen if the user is creating a folder named "example". Is that working?
- What will happen when the user is using the restet button to do a factory reset. Is a resinstall needed?

File: webserver/HuConJsonRpc.py (hunk 4 above)
Comment (JonasTrampe) [reply webserver/HuConJsonRpc.py]: As the file-system may give us the files in an unordered fashion sorting seems necessary.

File: webserver/HuConJsonRpc.py (hunk 3 above)
Comment (JonasTrampe) [reply webserver/HuConJsonRpc.py]: As the file-system may give us the files in an unordered fashion sorting seems necessary.

File: webserver/HuConJsonRpc.py [hunk 6]
```diff
@@ -229,7 +246,10 @@ def _load_file(self, rpc_request):
         """ Return the content of the file back to the browser.
         """
         try:
+            # TODO: Extract file base path (examples/user code) from rpc call and remove this hack
```
[the remark this answers is not in the document]
Comment (JonasTrampe) [reply webserver/HuConJsonRpc.py]: Yes adding a prefix for the file source would be beneficial.

[the remark this answers is not in the document]
Comment (JonasTrampe) [reply webserver/HuConJsonRpc.py]: Soll das in einer weiteren Aufgabe angegangen oder hier mit erledigt werden?

[the remark this answers is not in the document]
Comment (SaschaMzH) [reply webserver/HuConJsonRpc.py]: We can do a fix with a new task but I think we cannot release the image on this version.

**Comment by SaschaMzH** [comment]
[OnionFW bauen mit HuCon integriert](https://trello.com/c/2KneFH5U/9-onionfw-bauen-mit-hucon-integriert)

**Comment by SaschaMzH** [comment]
[Speicherort beim Image und bei der Installation auf das Home Verzeichnis legen.](https://trello.com/c/zwrCqxlE/19-speicherort-beim-image-und-bei-der-installation-auf-das-home-verzeichnis-legen)

**Review by SaschaMzH — APPROVED** [review]

**Pull request merged**